bgpd/bgp_nht.c

```diff
@@ -709,7 +709,24 @@ static void bgp_nht_ifp_table_handle(struct bgp *bgp,
 	bool nhop_other = false;
 
 	frr_each (bgp_nexthop_cache, table, bnc) {
-		if (bnc->ifindex_ipv6_ll != ifp->ifindex)
+		for (nhop = bnc->nexthop; nhop; nhop = nhop->next) {
+			if (nhop->ifindex == ifp->ifindex &&
+			    nhop->vrf_id == ifp->vrf->vrf_id)
+				nhop_found = true;
+			else
+				nhop_other = true;
+		}
+
+		if (up && !nhop_found && bnc->ifindex_ipv6_ll != ifp->ifindex)
+			/* up event cancelled if no nexthops available
+			 * for that interface */
+			continue;
+
+		if (!up && bnc->ifindex_ipv6_ll != ifp->ifindex &&
+		    (!nhop_found || nhop_other))
```

Comment:
in the down case, there are two cases with multiple nexthop:
- all interfaces are down => invalid the nexthop
- the interface is down but at least another is up => `continue;`

Could be check the status of the others interfaces ?

Reply:
a comment will be added